src/types/index.ts

```diff
@@ -4,12 +4,46 @@ export interface Webhook {
     webhookURL: string,
     transactionTypes: string[],
     accountAddresses: string[],
-    webhookType: WebhookType | null,
+    webhookType: WebhookType,
     authHeader: string
 }
 
-export type CreateWebhookRequest = Omit<Webhook, 'webhookID' & 'wallet'>;
-export type EditWebhookRequest = Omit<Webhook, 'webhookID' & 'wallet'>;
+export type CreateWebhookRequest = Omit<Webhook, 'webhookID' | 'wallet'>;
+export type EditWebhookRequest = Omit<Webhook, 'webhookID' | 'wallet'>;
+
+export interface CreateCollectionWebhookRequest {
+    collectionQuery: CollectionIdentifier
+    webhookURL: string,
+    transactionTypes: string[],
+    webhookType?: WebhookType,
+    authHeader?: string
+}
+
+export interface GetMintlistResponse {
+    result: MintlistObject[],
+    paginationToken: string
+}
+
+export interface MintlistObject {
+    mint: string,
+    name: string,
+}
+
+export type CollectionIdentifier = {
```

Comment:
You can actually make this a one_of type.

export type FirstCreatorIdentifier = { firstVerifiedCreators }
export type VerifiedCollectionIdentifier = { verifiedCollectionAddresses }
export type NftCollectionIdentifier = FirstCreatorIdentifier | VerifiedCollectionIdentifier